flowzone.yml

```diff
@@ -86,30 +87,22 @@
       set -ea
       # prevent git from existing with 141
       set +o pipefail
-      previous_tag="$(git --no-pager tag --list --sort=-version:refname "v*.*.*" --merged | head -n2 | tail -n1)"
-      release_notes_file="$(mktemp)"
-      git log ${previous_tag}..${{ github.event.pull_request.head.sha || github.event.head_commit.id }} --pretty=reference > "${release_notes_file}"
-      echo "file=${release_notes_file}" >> $GITHUB_OUTPUT
 
-  # FIXME: https://github.com/balena-io/deploy-to-balena-action/issues/194
-  - &updateBalenaReleaseNotes
-    name: Update balena release notes
-    run: |
-      set -ea
-      release_notes="$(cat < '${{ steps.release_notes.outputs.file }}' | jq -R -s .)"
-      app_id="$(curl --silent --retry 3 --fail \
-        "https://api.${{ inputs.balena_environment }}/v6/application?\$filter=slug%20eq%20%27${{ matrix.slug }}%27&\$select=id" \
-        -H 'Content-Type: application/json' \
-        -H 'Authorization: Bearer ${{ secrets.BALENA_API_KEY || secrets.BALENA_API_KEY_PUSH }}' \
-        | jq -r '.d[].id')"
-      release_id='${{ steps.balena_deploy.outputs.release_id }}'
-      if [[ -n $release_notes ]] && [[ -n $app_id ]] && [[ -n $release_id ]]; then
-          curl --silent --retry 3 --fail \
-            -X PATCH "https://api.${{ inputs.balena_environment }}/v6/release?\$filter=belongs_to__application%20eq%20${app_id}%20and%20id%20eq%20${release_id}" \
-            -H 'Content-Type: application/json' \
-            -H 'Authorization: Bearer ${{ secrets.BALENA_API_KEY || secrets.BALENA_API_KEY_PUSH }}' \
-            -d "{\"note\":${release_notes}}"
+      look_back=1
+      if [[ '${{ github.event.action }}' =~ closed ]] && [[ '${{ github.event.pull_request.merged }}' == true ]]; then
+          look_back=2
```

Comment:
this additional logic should select the correct range depending whether the release(s) are draft or finalised.

Reply:
I don't think we can hardcode that logic in this function as the caller is more aware of what is required to find the last tag for a diff.

For example, the `balena_publish` job (and others) are calling `*createLocalRefs` so the draft tag actually exists locally in those cases. It should behave similar to merged and look back 2 refs, shouldn't it?

What if we make `look_back: 2` an env var on this step, and calling jobs can override it if necessary?